controller/linodemachine_controller.go

```diff
@@ -429,6 +425,25 @@ func (r *LinodeMachineReconciler) addMachineToLB(
 	return nil
 }
 
+func (r *LinodeMachineReconciler) removeMachineFromLB(
+	ctx context.Context,
+	logger logr.Logger,
+	machineScope *scope.MachineScope,
+) error {
+	if machineScope.LinodeCluster.Spec.Network.LoadBalancerType != "dns" {
```

Comment:
Probably want to check explicitly that this is a NB type before calling DeleteNodeFromNB:
```suggestion
	if machineScope.LinodeCluster.Spec.Network.LoadBalancerType == "NodeBalancer" {
```
then on 438 check that it's "dns"

Reply:
Done